foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/version_specific/AbstractContainerizeMigration.java

```diff
@@ -0,0 +1,81 @@
+package com.boozallen.aissemble.upgrade.migration.version_specific;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import static org.apache.commons.lang3.StringUtils.repeat;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.PluginExecution;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import com.boozallen.aissemble.upgrade.migration.AbstractPomMigration;
+
+public abstract class AbstractContainerizeMigration extends AbstractPomMigration {
+
+    public static final String HABUSHU_MAVEN_PLUGIN_ID = "org.technologybrewery.habushu:habushu-maven-plugin";
+    private static final String FERMENTER_MDA_PLUGIN_ID = "org.technologybrewery.fermenter:fermenter-mda";
+    private static final String DOCKER_BUILD_TYPE = "docker-build";
+
+    protected boolean containsHabushuContainerizeGoal(MavenProject mavenProject) {
+        Plugin habushuPlugin = mavenProject.getPlugin(HABUSHU_MAVEN_PLUGIN_ID);
+        if (habushuPlugin != null) {
+            for (PluginExecution execution : habushuPlugin.getExecutions()) {
+                if (execution.getGoals().contains("containerize-dependencies")) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    protected boolean isDockerBuildPackage(MavenProject mavenProject) {
+        return StringUtils.equals(DOCKER_BUILD_TYPE, mavenProject.getModel().getPackaging());
+    }
+
+    protected boolean containsFermenterProfile(MavenProject mavenProject, String profileId) {
+        Plugin fermenterMdaPlugin = mavenProject.getPlugin(FERMENTER_MDA_PLUGIN_ID);
+        if (fermenterMdaPlugin != null) {
+            Xpp3Dom configuration = (Xpp3Dom) fermenterMdaPlugin.getConfiguration();
+
+            for (Xpp3Dom child: configuration.getChildren()) {
+                if(profileId.equals(child.getValue())) {
```

Comment:
A: could make this slightly more robust without much more complexity
```suggestion
                if("profile".equals(child.getName()) && profileId.equals(child.getValue())) {
```